### ISSUE TITLE
Automatic retries when sending fails

### ISSUE THREAD

**Issue by deibertf** [WPManageNinja/fluent-smtp#156]
Would be great if one could set up automatic retries. 
The Plugin could then try X times to send the mail and only after that mark the mail as "not sent".

**Comment by akerd** [comment]
I came here to ask the same question. It would be great to have a possibility to configure X retries after a sending fail.

**Comment by Krimoon** [comment]
This is a must feature, The number of failed sent emails is quite high!

**Comment by jpardenoy** [comment]
up

**Comment by felix-exon** [comment]
This would be super useful as I temporarily receive errors like this from my email server:

```
{
  "code": 422,
  "message": "SMTP Error: Could not authenticate.",
  "errors": [
    "SMTP Error: Could not authenticate."
  ]
}
```

A simple "send" then sends the mail without any issues, so a simple retry after x seconds would be a huge relief!

**Comment by Danmyrer** [comment]
Upvote this, I have the same Issue as @felix-exon

**Comment by RPWouters** [comment]
Up! Because this issue isnt fixed yet with o365

Server Response

{
  "code": 422,
  "message": "Unauthorized",
  "errors": [
    "Unauthorized"
  ]
}

**Comment by coenquant** [comment]
Yeah experience the same issue  can some one please makes it possible that Fluentsmtp setting allows resend  x time on failure

**Comment by dkiryakov** [comment]
Issue that was causing 422 for us seemed to happen only non-English versions of our website. I suspect WPML plugin interferes with the callback URLs generated by FluentSMTP. I added to Azure all possible URLs with possible language paths added by WPML and this seemed to resolve the issue. 

E.g., when your standard callback url is `https://mydomain.com/fluent-smtp/outlook_callback` make sure to add `https://mydomain.com/en/fluent-smtp/outlook_callback` , `https://mydomain.com/de/fluent-smtp/outlook_callback` and so on, for all possible language paths you have available to Azure's allowed callback URLs.
